### ISSUE TITLE
Tying toots to articles for providing a unified comments infrastructure

### ISSUE THREAD

**Issue by fbruetting** [tootsuite/mastodon#9030]
### Pitch

In principle, this would extent Mastodon from being a short message infrastructure to also being able to provide a unified way of handling comments to larger articles.

### Motivation

I always hate the fact, that we have to register at every news site, just for making a comment to one article (think of Disqus, etc.). This could easily be resolved by Mastodon with the benefit, that we also could better share articles throughout the fediverse by just boosts instead of links, and we’ll be able to mention all Mastodon users everywhere.

### Implementation thoughts

I think it would be good if new toots could be tied to articles an a mutual/bijective way. Toots should get a metadata entry for linking to an article, and articles should also refer to their “official” toot via a metadata entry.

A metadata entry in a toot would be reasonable, because this then would not reduce the available text content length and because the URL might need to be changed in future, if the article location changes. Being bejective leads to the advantage that then there can be one official toot, who’s thread could be shown via a widget below respective articles. This could help prevent stealing of news.

This issue overlaps a little bit with #6090. A commenting widget below an article could also handle the Mastodon API directly and just show the responses to the corresponding toot without the need of RSS at all. But I think that then depends on how someone wants to implement following further comments to articles.

**Comment by trwnh** [comment]
This is technically already possible with OAuth and with ActivityPub `Article` as with Plume blogs. I think that mostly depends on the blog author/software implementing an ActivityPub representation for comments, though.

**Comment by rugk** [comment]
I would love to see this:
* How it may be done: Like you can see it with _Discourse_, e.g. https://wordpress.org/plugins/wp-discourse/
* Also: It would be awesome to see that on https://blog.joinmastodon.org/, so I can just directly comment on Mastodon's blog articles. :smile:

**Comment by rugk** [comment]
> This is technically already possible with OAuth and with ActivityPub Article as with Plume blogs.

Well… yeah, I guess, however, that we are talking about a different use case/simpler implementation here.

What blog admins (especially of static sites!) want: Embed an iframe (maybe with come custom attributes or URL, so it knows what URL the toot or so is), so users can comment.

What may be needed here/could make it problematic: Admins can only point to one instance of Mastodon. Likely the one, where they have their own account and their own toot, that links to their blog.

Now for users to interact, however, they need to be redirected to their own instance. Currently, that is fortunately possible with a click on that icon and entering their Mastodon ID ([last step can be skipped with an add-on e.g.](https://addons.mozilla.org/de/firefox/addon/mastodon-simplified-federation/) :blush:).
However, what needs to be done here, is of course that the redirect does not open a new popup, but rather redirects inside of the iframe.

**Comment by fbruetting** [comment]
Yes, this is about an easy implementation of a federated comments network, so that we don’t have to register at every single site separately. The easier this would become, the more traction it will get.

Maybe it’s also possible for articles to be implicit toots. That means that you don’t have to always reference your own articles – the website/blog would then be the tooting instance (which may additionally reference the (private/personal) Mastodon accounts of all authors). So then there will be a Mastodon account for the site/blog, which posts the article (like RSS). Authors should then be able to opt in for being referenced by their articles, so that articles would be spread by both the website and the author.

When users subscribed to both accounts, they should not get that same article listed two times, but instead this should be combined into a single toot.

But I don’t know how this system works internally, these are just thoughts about how Mastodon’s interaction could be improved by a end user’s point of view. And I really want Mastodon to be able to replace RSS, because it seems to be about the same, and we get a federated comment infrastructure on top!
